Fixing compile time errors irt exception handling for Python 3. This …

Hypermanzer opened the pull request:
##### ISSUE TYPE
- Bugfix Pull Request
##### COMPONENT NAME

utilities/helper/accelerate
##### ANSIBLE VERSION

ansible-devel
##### SUMMARY

Fixing compile time errors irt exception handling for Python 3. This particular diff fixes problems with Exception handling and the use/misuse irt Python 3 of octal numbers (InvalidToken exceptions)

Commit:
Fixing compile time errors irt exception handling for Python 3. This particular diff fixes problems with Exception handling and the use/misues IRT Python 3 of octal numbers (InvalidToken exceptions).

gregdek commented:
Thanks @Hypermanzer for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

abadger commented:
This one will need to run on python2.4.  I gave @Hypermanzer the link to the python3 porting page: http://docs.ansible.com/ansible/developing_modules_python3.html which has recipes for how to modify the code to work for both 2.4 and python 3.

Commit:
1) Modified error handling to work with Python 2.4.
2) Fixed octal codes to fall in line with the ansible guide, "Porting Modules to Python 3"
3) updated the requirements.
All changes have been verified against Python 2.4, 2.6, & 3.5.

ghost commented:
1) Modified error handling to work with Python 2.4.
2) Fixed octal codes to fall in line with the ansible guide, "Porting Modules to Python 3"
3) updated the requirements to require 2.4 (not 2.6 as the doc originally stated)
All changes have been verified against Python 2.4, 2.6, & 3.5.

ghost commented:
Thank you. I'm working now on a fix now to make sure the patch supports 2.4 (and regresses properly against 2.6 and 3.5) in line with http://docs.ansible.com/ansible/developing_modules_python3.html.

ghost commented:
Re-tested & tests passed. 'ready_for_review'

gregdek commented:
Thanks @Hypermanzer. To the current maintainers, @jimi-c please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

Commit:
Corrected string of text...

Commit:
Missed umask octal conversion.

nitzmahone commented:
Good to merge once Travis is green.

sivel commented:
Same comment on this one too about py24 and use of `.format`.  At the moment due to the specific use of `.format` these changes require py27.

Commit:
corrected the string formatting (format isn't in 2.4 and even in some later versions of python 2.x requires an index). Tested for both compile ***& runtime exceptions*** against 2.4, 2.6, and 3.5.

Commit:
data type fix on format string

ghost commented:
ready_for_review

ghost commented:
@abadger Mind having a look?

abadger commented on utilities/helper/accelerate.py:
```diff
@@ -168,14 +168,15 @@ def daemonize_self(module, password, port, minutes, pid_file):
+    except OSError:
+        e       = get_exception()
+        message = "fork #1 failed: %d (%s)" % (e.errno, e.strerror)
+        module.fail_json(message)
```
@Hypermanzer I think we want to use message as a kwarg:  `module.fail_json(msg=message)`
@bcoca does this look right to you?  Logically, I think that it's correct.  If the fork fails then we're still in the first parent and so we want to use module.fail_json() to return json output.  But I don't write forking code very often so I'd appreciate a second set of eyes to make sure that's the correct thing to do here.

abadger commented on utilities/helper/accelerate.py:
```diff
@@ -187,9 +188,10 @@ def daemonize_self(module, password, port, minutes, pid_file):
+    except OSError:
+        e       = get_exception()
+        message = 'fork #2 failed: %d (%s)' % (e.errno, e.strerror)
+        module.fail_json(message)
```
By this point, I think we've already forked so we no longer want to return json on stdout.  So probably want to revert to the code that calls log() here.

abadger commented:
Made some comments, everything else looks okay to me.  @bcoca might correct me on one of the points.

Commit:
a) fail_json uses kwargs, b) outside the fork, log instead of returning json on stdout.

ghost commented:
Thanks, @abadger - ready_for_review

bcoca commented on utilities/helper/accelerate.py:
```diff
@@ -685,11 +691,16 @@ def main():
-            except OSError, e:
+            except OSError:
+                e        = get_exception()
+                message  = 'the accelerate daemon appears to be running'
```
not really a reason to avoid merge, but it would be better to compose the message inside the if (only when we use it)

abadger commented:
This is ready to merge.  I'll be on a train today so if someone else wants to do it, that's fine or I can do it tomorrow.

shipit
